lib/autokey/qtui/dialogs/show_recent_script_errors.py

```diff
@@ -0,0 +1,152 @@
+            self.clear_error_list()
+
+    @pyqtSlot()
+    def update_and_show(self):
+        error_count = len(self.recent_script_errors)
+        if error_count:
+            if self.currently_viewed_error_index >= error_count:
+                self.currently_viewed_error_index = error_count-1
+
+            self.has_next_error.emit(self.currently_viewed_error_index < error_count-1)
+            self.has_previous_error.emit(self.currently_viewed_error_index > 0)
+
+            logger.info("User views the last script errors. There are {} errors to review.".format(error_count))
+            self._show_error(self.currently_viewed_error_index)
+
+            self.show()
+        else:
+            logger.error(
+                "User is able to view the script error dialogue, even if no errors are available. "
+                "This should be impossible. Do not show the dialogue window.")
+
+    @pyqtSlot()
+    def show_next_error(self):
+        """
+        Switch to the next error in the error list.
+        The connection from the Next button to this slot function is defined in the .ui file.
+        """
+        logger.debug("User views the next error.")
+        # Out of bounds handling is not needed, as the Next button gets disables when the last error is reached.
+        # See has_next_error Signal.
+        self.currently_viewed_error_index += 1
+        self.has_next_error.emit(self.currently_viewed_error_index < len(self.recent_script_errors)-1)
+        self._show_error(self.currently_viewed_error_index)
+
+    @pyqtSlot()
+    def show_previous_error(self):
+        """
+        Switch to the previous error in the error list.
+        The connection from the Previous button to this slot function is defined in the .ui file.
+        """
+        logger.debug("User views the previous error.")
+        # Out of bounds handling is not needed, as the Previous button gets disables when the first error is reached.
+        # See has_previous_error Signal.
+        self.currently_viewed_error_index -= 1
+        self.has_previous_error.emit(self.currently_viewed_error_index > 0)
+        self._show_error(self.currently_viewed_error_index)
+
+    def remove_currently_shown_error_from_error_list(self):
+        """
+        Delete the currently shown error from the error list.
+        Shows the next error, if available. Otherwise, show the previous error, if available.
+        Or clear the list and hide the window, if the deleted error was the only one in the list.
+        """
+        error_count = len(self.recent_script_errors)
+        if error_count == 1:
+            self.clear_error_list()
+        else:
+            del self.recent_script_errors[self.currently_viewed_error_index]
+            if self.currently_viewed_error_index == error_count-1:
+                self.show_previous_error()
+            else:
+                self.has_next_error.emit(self.currently_viewed_error_index < len(self.recent_script_errors)-1)
```

Comment:
It has a similar structure to the update_and_show() function.

Reply:
Sorry, this is my first code review. I was referring only to line 132, which is same as 80/81. I didn't expect the whole block to come with it

Reply:
> Sorry, this is my first code review. 

It’s my second :)


> I was referring only to line 132, which is same as 80/81. I didn't expect the whole block to come with it

The GitHub interface seems to be a bit ambiguous here. So I interpreted that two-words comment together with the full block quoted by GitHub as actually referring to the full block quoted. And your comment is technically correct for the full block, too, as the logic flow is kind of similar. So that didn’t clear the ambiguity, either.